src/app/records/PresentationNodeLeaf.tsx

```diff
@@ -13,17 +15,103 @@ export default function PresentationNodeLeaf({
   ownedItemHashes,
   completedRecordsHidden,
   redactedRecordsRevealed,
+  sortRecordProgression,
 }: {
   node: DimPresentationNodeLeaf;
   ownedItemHashes?: Set<number>;
   completedRecordsHidden: boolean;
   redactedRecordsRevealed: boolean;
+  sortRecordProgression: boolean;
 }) {
+  const newNode = { ...node };
```

Comment:
You don't need to copy it until you need to copy it (i.e. if `sortRecordProgression` is `true`.